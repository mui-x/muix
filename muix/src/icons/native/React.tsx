import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)