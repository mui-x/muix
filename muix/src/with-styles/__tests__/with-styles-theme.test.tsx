/** @jsx platform.createElement */
import React from 'react'

import { TSheeter } from 'reactxx-typings'
import { platform } from 'reactxx-sheeter'

import { ts, traceComponentEx, Shape, ReactAny, theme, ThemeProvider } from "reactxx-tests"

describe("WITH STYLES THEME", () => {
  it("WEB", () => traceTest(true));
  it("NATIVE", () => traceTest(false));
});

const customTheme: TSheeter.getTheme<Shape> = {
  ...theme,
  secondary: {
    ...theme.secondary,
    normal: {
      color: 'lightcyan',
      background: 'black',
    },
  },
}

const traceTest = (isWeb: boolean) => {

  traceComponentEx(isWeb, 3,

    theme => ({
      label: [
        theme.secondary.normal,
        {
          fontWeight: 'bold'
        }
      ],
      root: {},
      webOnly: {},
      nativeOnly: {},
    }),

    ({ classes, toClassNames, children }) => {
      return <ReactAny classNameX={toClassNames([classes.label])}>{children}</ReactAny>
    },

    Comp => <ReactAny>
      <Comp>Default Theme</Comp>
      <ThemeProvider theme={customTheme}>
        <Comp>Custom Theme</Comp>
      </ThemeProvider>
    </ReactAny>
  )
}
