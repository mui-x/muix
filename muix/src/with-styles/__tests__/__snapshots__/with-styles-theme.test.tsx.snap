// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLES THEME NATIVE 1`] = `
<div>
  <div
    data="REACTANY"
  >
    <div
      data="REACTANY"
      style="font-weight: bold; color: black;"
      trace="
fontWeight: bold /*TestComponent.sheet.label[1]*/
backgroundColor: lightblue /*TestComponent.sheet.label[0]*/
color: black /*TestComponent.sheet.label[0]*/"
    >
      Default Theme
    </div>
    <div
      data="REACTANY"
      style="font-weight: bold; background-color: blue; color: black;"
      trace="
fontWeight: bold /*TestComponent.sheet.label[1]*/
backgroundColor: blue /*TestComponent.sheet.label[0]*/
color: black /*TestComponent.sheet.label[0]*/"
    >
      Custom Theme
    </div>
  </div>
</div>
`;

exports[`WITH STYLES THEME WEB 1`] = `
<div>
  <div
    data="REACTANY"
  >
    <div
      class="c b a"
      data="REACTANY"
      trace="
.c { font-weight:bold /*TestComponent.sheet.label[1]*/ }
.b { background-color:lightblue /*TestComponent.sheet.label[0]*/ }
.a { color:black /*TestComponent.sheet.label[0]*/ }"
    >
      Default Theme
    </div>
    <div
      class="c d a"
      data="REACTANY"
      trace="
.c { font-weight:bold /*TestComponent.sheet.label[1]*/ }
.d { background-color:blue /*TestComponent.sheet.label[0]*/ }
.a { color:black /*TestComponent.sheet.label[0]*/ }"
    >
      Custom Theme
    </div>
  </div>
</div>
`;
